README.md

```diff
@@ -1,6 +1,19 @@
 ## AWS C S3
 
-C99 library implementation for communicating with the S3 service, designed for maximizing throughput on high bandwidth EC2 instances.
+The AWS-C-S3 library is an asynchronous AWS S3 client focused on maximizing throughput and network utilization.
+
+### Key features:
+- **Automatic Request Splitting**: Improves throughput by automatically splitting the request into part-sized chunks and performing parallel uploads/downloads of these chunks over multiple connections.
+- **Enhanced Transfer Reliability**: Increases resilience against network failures by retrying individual failed chunks of a file transfer, eliminating the need to restart transfers from scratch.
+- **DNS Load Balancing**: DNS resolver continuously harvests Amazon S3 IP addresses to distribute connections across the fleet, significantly boosting throughput and enhancing overall transfer efficiency.
+- **Advanced Network Management**: The client incorporates automatic request parallelization, effective timeouts and retries, and efficient connection reuse. This approach helps to maximize throughput and network utilization, and to avoid network overloads.
+- **Thread Pools and Async I/O**: Avoids bottlenecks associated with single-thread processing.
+- **Parallel Reads**: Supports parallel file reads from disk when uploading a file to S3.
```

Comment:
again, explaining why, since it's not obvious that parallel reads would be faster
```suggestion
- **Parallel Reads**: When uploading a large file from disk, reads from multiple parts of the file in parallel. This is faster than reading the file sequentially from beginning to end.
```